### PR TITLE
Adds PATCH /api/articles/:article_id API

### PR DESCRIPTION
**Pull request by cdesale**
- Adds PATCH /api/articles/:article_id API
- Adds necessary tests

### CODE REVIEW

File: __tests__/patch-article-by-id-api.test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  test("404: responds with an error msg if invalid inc_votes value", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: "invalid" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("201: responds with a updated article with a given article id", () => {
```
Comment (ghost) [inline __tests__/patch-article-by-id-api.test.js]: Have we created something or have we updated something? 201 is for new created content. Here a 200 would be more appropriate.

File: __tests__/patch-article-by-id-api.test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  test("404: responds with an error msg if article does not exist", () => {
+    return request(app)
+      .patch("/api/articles/999")
+      .send({ inc_votes: 1 })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
+  test("404: responds with an error msg if invalid inc_votes value", () => {
```
Comment (ghost) [inline __tests__/patch-article-by-id-api.test.js]: Watch that test description, its not a 404 in your test.

File: __tests__/patch-article-by-id-api.test.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: expect.any(String),
+          votes: 0,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
```
Comment (ghost) [inline __tests__/patch-article-by-id-api.test.js:85]: Just a missing test to look at what happens if patching to a bad id, so patch /api/articles/banana Your error handler will probably already catch it, but good to have a solid test for it.

File: controllers/article-controller.js [hunk 4]
```diff
@@ -17,3 +17,18 @@ exports.getArticles = (req, res, next) => {
     })
     .catch(next);
 };
+
+exports.patchArticle = (req, res, next) => {
+  const { article_id } = req.params;
+  const { inc_votes } = req.body;
+
+  if (isNaN(article_id) || isNaN(inc_votes)) {
```
Comment (ghost) [inline controllers/article-controller.js]: This should not be a custom reject, PSQL has very specific error codes for this situation that you should be looking out for in your error handlers.

File: controllers/article-controller.js [hunk 5]
```diff
@@ -17,3 +17,18 @@ exports.getArticles = (req, res, next) => {
...
+  const { article_id } = req.params;
+  const { inc_votes } = req.body;
+
+  if (isNaN(article_id) || isNaN(inc_votes)) {
+    return res.status(400).send({ msg: "Bad request" });
+  }
+
+  updateArticleById(article_id, inc_votes)
+    .then((data) => {
+      res.status(201).send({ article: data });
```
Comment (ghost) [inline controllers/article-controller.js]: Is this a 201?

File: models/article-model.js [hunk 6]
```diff
@@ -30,6 +30,27 @@ GROUP BY
...
+      RETURNING *
+    `;
+  return db
+    .query(articleExistsQuery, [article_id])
+    .then(({ rows }) => {
+      if (!rows[0].exists) {
+        return Promise.reject({ status: 404, msg: "not found" });
+      }
+
+      return db.query(articleUpdateQuery, [inc_votes, article_id]);
```
Comment (ghost) [inline models/article-model.js]: We should be able to do this as one db query. Consider a check exists function that would be resuable. You may need a promise all to achieve it.

**Review by ghost — CHANGES_REQUESTED** [review]
A few areas to look at here. Take a look at the comments and please resubmit this pr. It also looks like the endpoints.json has not been updated for this either.

File: app.js [hunk 7]
```diff
@@ -5,6 +5,11 @@ const { getAvailableEndpoints } = require("./controllers/document-controller");
 const { getArticles } = require("./controllers/article-controller");
 const { getArticle } = require("./controllers/article-controller");
 const { getAllComments } = require("./controllers/comment-controller");
+const { patchArticle } = require("./controllers/article-controller");
+<<<<<<< HEAD
+const { postComments } = require("./controllers/comment-controller");
+=======
+>>>>>>> 8895b57 (Adds PATCH /api/articles/:article_id API)
```
Comment (nc-jonathan-mcguire) [inline app.js:13]: Think you'll want this merge comment stuff removing

File: controllers/comment-controller.js [hunk 8]
```diff
@@ -1,5 +1,24 @@
-const { selectCommentsByArticleId } = require("../models/comment-model");
+const {
+<<<<<<< HEAD
+  selectCommentsByArticleId,
+  inserCommentsForArticleByArticleId,
+} = require("../models/comment-model");
 
+=======
```
Comment (nc-jonathan-mcguire) [inline controllers/comment-controller.js]: Think you'll want this merge comment stuff removing

File: models/comment-model.js [hunk 9]
```diff
@@ -19,3 +19,34 @@ ORDER BY created_at DESC;`,
       return rows;
     });
 };
+<<<<<<< HEAD
+
+exports.inserCommentsForArticleByArticleId = (article_id, username, body) => {
+=======
+exports.insertCommentsForArticleByArticleId = (article_id, username, body) => {
+>>>>>>> 4c590a7 (Addressed typo error in comment-model)
```
Comment (nc-jonathan-mcguire) [inline models/comment-model.js]: Think you'll want this merge comment stuff removing

**Review by nc-jonathan-mcguire — APPROVED** [review]
Looks okay 👍 Think you'll want this merge comment stuff removing in places, but ive left comments

**Pull request merged**